src/mgmt/srv_pool.c

```diff
@@ -378,15 +378,25 @@ ds_mgmt_pool_query(uuid_t pool_uuid, d_rank_list_t *svc_ranks, d_rank_list_t **r
 		   daos_pool_info_t *pool_info, uint32_t *pool_layout_ver,
 		   uint32_t *upgrade_layout_ver)
 {
+	uint64_t deadline;
+
 	if (pool_info == NULL) {
 		D_ERROR("pool_info was NULL\n");
 		return -DER_INVAL;
 	}
 
 	D_DEBUG(DB_MGMT, "Querying pool "DF_UUID"\n", DP_UUID(pool_uuid));
 
-	return ds_pool_svc_query(pool_uuid, svc_ranks, ranks, pool_info,
-				 pool_layout_ver, upgrade_layout_ver);
+	/*
+	 * Use a fixed timeout that matches what the control plane uses for the
+	 * moment.
+	 *
+	 * TODO: Pass the deadline from dmg (or daos_server).
+	 */
+	deadline = daos_getmtime_coarse() + 5 * 60 * 1000;
```

Comment:
is the control plane deadline for pool query 5 minutes? I thought I had seen in some command outputs dmg pool query timing out after 3 minutes.

Reply:
It appeared so when I tested it recently. I have, however, lost track of what master commit I was using.

```
[liw2@boro-66 daos]$ time dmg -o ~/daos_control.yml pool query test-pool-0
ERROR: dmg: pool query failed: client: code = 510 description = "the *control.PoolQueryReq request timed out after 5m0s"
ERROR: dmg: client: code = 510 resolution = "retry the request or check server logs for more information"

real    5m0.046s
user    0m0.026s
sys     0m0.032s
```

Reply:
My reading of the current base commit 008a9ba7863af29edc24ebd1780aa0e63ea32746 is that the dmg pool query timeout is indeed 5 minutes (the Go code responsible for this value is not so clear though):

```
	// DefaultPoolTimeout is the default timeout for a pool request.
	DefaultPoolTimeout = 5 * time.Minute
```

```
	// Start with this value... If anything takes longer than this,
	// it almost certainly needs to be fixed.
	defaultRequestTimeout = 5 * time.Minute
```

Reply:
Thanks. I was being lazy and only looking at the description of the issue in ticket https://daosio.atlassian.net/browse/DAOS-12976 that was filed based on a (several months older) version of master.

Reply:
No problem. It did take me almost 20 minute to find out what in the code controls the timeout.

Reply:
IIRC, the 5m time out in question is a client/dmg side timeout passed in a context when calling into gRPC API for the control service.